### PR TITLE
Fixed some things. Started adding comments

### PR DESCRIPTION
**Pull request by mfrlin**
I've fixed some minor things and started adding comments. Would appreciate some feedback before I loose too much time heading in wrong direction. 

Can't figure out how to trigger addComment() when submiting a comment via generated form. Can't figure out how to trigger deletePost() via delete_link. JavaScript help desired :)

### CODE REVIEW

**Comment by mitar** [comment]
Please update from main repository. This pull request cannot be merged automatically.

File: piplmesh/account/backends.py [hunk 1]
```diff
@@ -75,9 +75,9 @@ def authenticate(self, facebook_access_token, request):
             # TODO: Do we know if all e-mail addresses given by Facebook are verified?
             # TODO: Does not Facebook support multiple e-mail addresses? Which one is given here?
             user.email = facebook_profile_data.get('email') or None
-        if user.gender is None:
-            # TODO: Does it really map so cleanly?
-            user.gender = facebook_profile_data.get('gender') or None
+        
+        user.gender = facebook_profile_data.get('gender') or None
```
Comment (mitar) [inline piplmesh/account/backends.py]: Why this change?

File: piplmesh/account/fields.py [hunk 2]
```diff
@@ -72,6 +72,7 @@ def __init__(self, *args, **kwargs):
 
 class GenderField(mongoengine.StringField):
     def __init__(self, *args, **kwargs):
+        # TODO; Do we want to default gender to female? Wouldn't it be nicer to None?
```
Comment (mitar) [inline piplmesh/account/fields.py]: Use `:` and not `;`.

Good TODO to ask.

Comment (mitar) [inline piplmesh/account/fields.py]: But none == not required. So the question is if want this field to be required or not.

File: piplmesh/api/models.py [hunk 3]
```diff
@@ -41,6 +43,8 @@ class Post(base.AuthoredDocument):
     def save(self, *args, **kwargs):
         self.updated_time = timezone.now()
         return super(Post, self).save(*args, **kwargs)
+    
+
```
Comment (mitar) [inline piplmesh/api/models.py:49]: What are those new lines?

File: piplmesh/frontend/static/piplmesh/js/home.js [hunk 4]
```diff
@@ -1,5 +1,6 @@
 var POSTS_LIMIT = 20;
-var POSTS_DATE_UPDATE_INTERVAL = 60000; // ms
+var POSTS_DATE_UPDATE_INTERVAL = 60000;
+// ms
```
Comment (mitar) [inline piplmesh/frontend/static/piplmesh/js/home.js]: Why newline?

File: piplmesh/api/models.py [hunk 5]
```diff
@@ -32,6 +32,8 @@ class Post(base.AuthoredDocument):
 
     comments = mongoengine.ListField(mongoengine.EmbeddedDocumentField(Comment), default=lambda: [], required=False)
     attachments = mongoengine.ListField(mongoengine.EmbeddedDocumentField(Attachment), default=lambda: [], required=False)
+    
+    is_edited = mongoengine.BooleanField(default=False)
```
Comment (mitar) [inline piplmesh/api/models.py]: What is this?

File: piplmesh/frontend/static/piplmesh/js/home.js [hunk 6]
```diff
@@ -31,7 +32,7 @@ function initializeEmptyColumnsForPanels() {
 function orderPanelsDefault() {
     var numberOfColumns = howManyColumns();
 
-    $('.panel').each(function (index, panel) {
+    $('.panel').each(function(index, panel) {
```
Comment (mitar) [inline piplmesh/frontend/static/piplmesh/js/home.js]: No, anonymous functions should have space after `function` part. [See our style example](https://github.com/wlanslovenija/PiplMesh/pull/199#commitcomment-1707793).

File: piplmesh/frontend/static/piplmesh/js/home.js [hunk 7]
```diff
@@ -54,23 +55,22 @@ function sendOrderOfPanelsToServer() {
 
     if (numberOfColumns) {
         $.post(URLS.panels_order, {
-            'names': names,
-            'columns': columns,
-            'number_of_columns': numberOfColumns
+            'names' : names,
```
Comment (mitar) [inline piplmesh/frontend/static/piplmesh/js/home.js]: We don't have space before `:`. Why space?

File: piplmesh/frontend/static/piplmesh/js/home.js [hunk 8]
```diff
@@ -54,23 +55,22 @@ function sendOrderOfPanelsToServer() {
...
-    }, function (data, textStatus, jqXHR) {
+        'number_of_columns' : howManyColumns()
+    }, function(data, textStatus, jqXHR) {
         if (data.length == 0) {
             orderPanelsDefault();
-        }
-        else {
-            $.each(data, function (i, column) {
-                $.each(column, function (j, panel) {
+        } else {
```
Comment (mitar) [inline piplmesh/frontend/static/piplmesh/js/home.js]: We do not have `else` on the same line as `}`. Why this change?

File: piplmesh/frontend/static/piplmesh/js/home.js [hunk 9]
```diff
@@ -144,80 +148,107 @@ function Post(data) {
...
+
+        return comments;
+
+    }
+
+    function addComment() {
+        var message = $('#comment_text').val();
+        $.ajax({
+            'type' : 'POST',
+            'url' : '/api/v1/post/' + self.id + '/comments/',
```
Comment (mitar) [inline piplmesh/frontend/static/piplmesh/js/home.js]: Don't have hard-coded URLs.

**Comment by mitar** [comment]
OK. Why code style changes? This is not really something you do in collaborative project where there is already established code style. So please explain and/or change back.

File: piplmesh/account/backends.py (hunk 1 above)
Comment (mfrlin) [inline piplmesh/account/backends.py]: because user gender defaults to female and doesnt take into consideration facebook gender because of the fi statement
if user.gender is None:

File: piplmesh/frontend/static/piplmesh/js/home.js (hunk 4 above)
Comment (mfrlin) [inline piplmesh/frontend/static/piplmesh/js/home.js]: i tried the format function to get rid of the tabs in the code and it went south. ill fix those empty lines and newlines, didn't have the time yesterday.

File: piplmesh/api/models.py (hunk 5 above)
Comment (mfrlin) [inline piplmesh/api/models.py]: If post is edited this will switch to true and show when is displayed (similar to facebook without message history)
that is if we will enable edit options. i thought its a good idea.

File: piplmesh/frontend/static/piplmesh/js/home.js (hunk 6 above)
Comment (mfrlin) [inline piplmesh/frontend/static/piplmesh/js/home.js]: formatting did this. would be nice if i could revert to previous commit somehow?

File: piplmesh/frontend/static/piplmesh/js/home.js (hunk 7 above)
Comment (mfrlin) [inline piplmesh/frontend/static/piplmesh/js/home.js]: formatting

File: piplmesh/frontend/static/piplmesh/js/home.js (hunk 8 above)
Comment (mfrlin) [inline piplmesh/frontend/static/piplmesh/js/home.js]: formatting, i need to revert it to previous commit.

File: piplmesh/frontend/static/piplmesh/js/home.js (hunk 6 above)
Comment (mitar) [inline piplmesh/frontend/static/piplmesh/js/home.js]: Of course you can. Read git documentation/tutorials how to do it.

File: piplmesh/api/models.py (hunk 5 above)
Comment (mitar) [inline piplmesh/api/models.py]: But it is unneeded until we have edit functionality. Why adding it now to the model?

And probably we want to keep history anyway, if we want to do it. So this could be a separate issue.

**Comment by mitar** [comment]
Please update from main repository.

File: piplmesh/account/backends.py (hunk 1 above)
Comment (mitar) [inline piplmesh/account/backends.py]: Yes. But the idea is that if user has already set some value, then we assume that she wants that value, even if it is different than Facebook's.

Probably the right solution here is to not make gender required and to have it assigned none value for default.

**Comment by mitar** [comment]
Please update from main repository to be able to merge.

**Comment by mitar** [comment]
OK. You should really learn how to merge and work with the repository.

This pull request became garbage. I am closing it. Please open new one, clean, and without those bad commits.

Also, in each pull request concentrate only on one thing. So, fixing some things should be one pull request. Comments another.

It is nothing wrong if pull request is not finished when you start it, make it to the point to which you know to do it and then ask for help. But this currently is simply unreadable so I do not even know what is wrong with it or where you have problems.
